### PR TITLE
Close #2093 : better ClusteredDedicated representation

### LINKED ISSUE #2093: HumandReadable representation of ClusteredDedicated too limited
Because DedicatedClusteredResourcePoolImpl does not implement HumanReadable, it ends up showing very limited info on HumanReadable string about the clustered dedicated tier, for example : 

```
                clustered-dedicated:
                    size: 2 MB (persistent)
                    tierHeight: 10
```

We can better than that, for example : 
```
                clustered-dedicated:
                    size: Pool{ 2MB clustered-dedicated(persistent) from='primary-server-resource'}
                    tierHeight: 10
```

### PR DESCRIPTION
**Pull request by anthonydahanne**
* also tweak the test to test ClusteredShared

### CODE REVIEW

**Comment by anthonydahanne** [comment]
@ljacomet @henri-tremblay this can wait next release cycle, post 3.4

File: clustered/integration-test/src/test/resources/clusteredConfiguration.txt [hunk 1]
```diff
@@ -14,7 +28,7 @@ caches:
                     size: 1 MB
                     tierHeight: 1000
                 clustered-dedicated:
-                    size: 2 MB (persistent)
+                    size: Pool{ 2MB clustered-dedicated(persistent) from='primary-server-resource'}
```
Comment (ljacomet) [inline clustered/integration-test/src/test/resources/clusteredConfiguration.txt]: A clustered resource will always be marked as `persistent`, should this be suppressed from the `toString()` and `readableString()`?

Comment (anthonydahanne) [reply clustered/integration-test/src/test/resources/clusteredConfiguration.txt]: are you talking about removing 3 lines from : https://github.com/ehcache/ehcache3/blob/a25d02c4dcb5744c6403447237a23e7ac93c7e01/clustered/client/src/main/java/org/ehcache/clustered/client/internal/config/DedicatedClusteredResourcePoolImpl.java#L83 ?

Comment (ljacomet) [reply clustered/integration-test/src/test/resources/clusteredConfiguration.txt]: yes

Comment (anthonydahanne) [reply clustered/integration-test/src/test/resources/clusteredConfiguration.txt]: ok

**Comment by anthonydahanne** [comment]
@ljacomet please reconsider

**Pull request merged**